contrib/mesos/pkg/scheduler/scheduler.go

```diff
@@ -449,7 +449,7 @@ func (k *KubernetesScheduler) StatusUpdate(driver bindings.SchedulerDriver, task
 func (k *KubernetesScheduler) reconcileTerminalTask(driver bindings.SchedulerDriver, taskStatus *mesos.TaskStatus) {
 	task, state := k.taskRegistry.UpdateStatus(taskStatus)
 
-	if (state == podtask.StateRunning || state == podtask.StatePending) && taskStatus.SlaveId != nil &&
+	if (state == podtask.StateRunning || state == podtask.StatePending) &&
```

Comment:
I asked about the `Source` and `Reason` because the `if` block that you modified checks for some very specific values there. You also removed the `SlaveId` check, which I still can't decide if I'm concerned about: that may have been there for a good reason.